From 1008fa50a48afeb2922f3300f9f420f581eae9bb Mon Sep 17 00:00:00 2001
From: Jeremy Johnson <jeremy.johnson@arm.com>
Date: Thu, 30 May 2019 10:52:40 +0100
Subject: [PATCH] Turn off GCC format-overflow checking for xtest

With the OpenEmbedded change to use GCC9, this introduces format-overflow
checking by default. The test code in xtest shows up as a false positive
for this check and warnings are treated as errors. This change disables
the format-overflow checking.

Note: Fixing the code so that the warning does
not occur would entail changing the key generated in the test.
---
 host/xtest/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/host/xtest/Makefile b/host/xtest/Makefile
index f226500..6170df1 100644
--- a/host/xtest/Makefile
+++ b/host/xtest/Makefile
@@ -155,7 +155,8 @@ CFLAGS += -Wall -Wcast-align -Werror \
 	  -Wshadow -Wstrict-prototypes -Wswitch-default \
 	  -Wwrite-strings \
 	  -Wno-declaration-after-statement \
-	  -Wno-missing-field-initializers -Wno-format-zero-length
+	  -Wno-missing-field-initializers -Wno-format-zero-length \
+	  -Wformat-overflow=0
 endif
 
 CFLAGS += -g3
